# Difference in Differences

## Study Case 1 - New Jersey vs Pennsylvania

In [11]:
#impoprt packages

import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.impute import SimpleImputer
import statsmodels.api as smf
%matplotlib inline



In [3]:
#import data

df = pd.read_csv('njmin3.csv')

df.head()

NJ  POST_APRIL92  NJ_POST_APRIL92    fte  bk  kfc  roys  wendys  co_owned  \
0   1             0                0  15.00   1    0     0       0         0   
1   1             0                0  15.00   1    0     0       0         0   
2   1             0                0  24.00   0    0     1       0         0   
3   1             0                0  19.25   0    0     1       0         1   
4   1             0                0  21.50   1    0     0       0         0   

   centralj  southj  pa1  pa2   demp  
0         1       0    0    0  12.00  
1         1       0    0    0   6.50  
2         1       0    0    0  -1.00  
3         0       0    0    0   2.25  
4         0       0    0    0  13.00

### Cleaning & Wrangling

In [4]:
# null values
df.isna().sum()

NJ                  0
POST_APRIL92        0
NJ_POST_APRIL92     0
fte                26
bk                  0
kfc                 0
roys                0
wendys              0
co_owned            0
centralj            0
southj              0
pa1                 0
pa2                 0
demp               52
dtype: int64

In [6]:
df.dtypes

NJ                   int64
POST_APRIL92         int64
NJ_POST_APRIL92      int64
fte                float64
bk                   int64
kfc                  int64
roys                 int64
wendys               int64
co_owned             int64
centralj             int64
southj               int64
pa1                  int64
pa2                  int64
demp               float64
dtype: object

In [7]:
missingvalues = SimpleImputer(missing_values = np.nan,
                              strategy = 'mean').fit(df[['fte', 'demp']])

In [8]:
df[['fte', 'demp']] = missingvalues.transform(df[['fte', 'demp']])

In [9]:
df.isna().sum()

NJ                 0
POST_APRIL92       0
NJ_POST_APRIL92    0
fte                0
bk                 0
kfc                0
roys               0
wendys             0
co_owned           0
centralj           0
southj             0
pa1                0
pa2                0
demp               0
dtype: int64

### Model

In [10]:
# isolate variables X and Y
X = df.iloc[:,0:3].values
y = df.iloc[:, 3].values

In [14]:
X = smf.add_constant(X)
model1 = smf.OLS(y,X).fit()
model1.summary(yname = 'FTE',
               xname = ('intercept',
                        'New Jersey',
                        'After April 1992',
                        'New Jersey and after April 1992')           
              )

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    FTE   R-squared:                       0.007
Model:                            OLS   Adj. R-squared:                  0.004
Method:                 Least Squares   F-statistic:                     1.974
Date:                Fri, 03 Mar 2023   Prob (F-statistic):              0.116
Time:                        09:14:36   Log-Likelihood:                -2986.2
No. Observations:                 820   AIC:                             5980.
Df Residuals:                     816   BIC:                             5999.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
===================================================================================================
                                      coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------
intercept                          23.2728      1.041     22.349      0.000      21.229      25.317
New Jersey                         -2.8157      1.159     -2.430      0.015      -5.091      -0.541
After April 1992                   -2.1108      1.473     -1.433      0.152      -5.001       0.780
New Jersey and after April 1992     2.6810      1.639      1.636      0.102      -0.536       5.898
==============================================================================
Omnibus:                      232.659   Durbin-Watson:                   1.847
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              908.337
Skew:                           1.289   Prob(JB):                    5.72e-198
Kurtosis:                       7.465   Cond. No.                         11.4
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

***New Jersey and April 92*** coef show us that minimum wage increase has positive impact on employment (2.68)